### PR TITLE
Add support to export & import Kibana objects

### LINKED ISSUE #69: Support export/import of Kibana objects
Docs https://docs.logz.io/api/#operation/exportSavedObjects

### PR DESCRIPTION
**Pull request by pengux**
Docs: https://docs.logz.io/api/#operation/exportSavedObjects

Closes #69

### CODE REVIEW

**Comment by yyyogev** [comment]
I fixed the Travis config, the checks should pass now on your branch. Just fix the conflicting file

**Comment by pengux** [comment]
> I fixed the Travis config, the checks should pass now on your branch. Just fix the conflicting file

Thanks, I've resolved the conflicts now.

**Comment by yyyogev** [comment]
Great! We'll review it shortly.

File: kibana_objects/client_kibana_objects_import.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		return nil, fmt.Errorf("%d %s", resp.StatusCode, jsonBytes)
+	}
+
+	results := &ImportResults{}
+	err = json.Unmarshal(jsonBytes, results)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal response body into ImportResults: %w", err)
+	}
+
+	return results, nil
```
Comment (yyyogev) [inline kibana_objects/client_kibana_objects_import.go:70]:
```suggestion
	var results ImportResults
	err = json.Unmarshal(jsonBytes, &results)
	if err != nil {
		return nil, fmt.Errorf("could not unmarshal response body into ImportResults: %w", err)
	}

	return &results, nil
```

File: kibana_objects/client_kibana_objects.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// user's logz.io API token and API base URL
+func New(apiToken string, baseUrl string) (*KibanaObjectsClient, error) {
+	if len(apiToken) == 0 {
+		return nil, fmt.Errorf("API token not defined")
+	}
+	if len(baseUrl) == 0 {
+		return nil, fmt.Errorf("Base URL not defined")
+	}
+
+	c := &KibanaObjectsClient{
```
Comment (yyyogev) [inline kibana_objects/client_kibana_objects.go:31]: please use proper naming, client instead of c, etc..

**Review by yyyogev — CHANGES_REQUESTED** [review]
Great work!
some small changes

**Comment by yyyogev** [comment]
@pengux have you finished with the changes?

**Comment by pengux** [comment]
> @pengux have you finished with the changes?

I'll try to finish the last changes this week.

File: kibana_objects/client_kibana_objects_export.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+type ExportResults struct {
+	KibanaVersion string                   `json:"kibanaVersion"`
+	Hits          []map[string]interface{} `json:"hits"`
+}
+
+// Export allows export of the Kibana objects configuration.
+// https://docs.logz.io/api/#operation/exportSavedObjects
+func (c *KibanaObjectsClient) Export(t exportType) (*ExportResults, error) {
+	exportPayload := struct {
```
[the remark this answers is not in the document]
Comment (pengux) [reply kibana_objects/client_kibana_objects_export.go:35]: Addressed in 4577f0a

File: kibana_objects/client_kibana_objects.go (hunk 2 above)
Comment (pengux) [reply kibana_objects/client_kibana_objects.go]: Addressed in ecdf91d

**Comment by pengux** [comment]
@yyyogev I've addressed all the comments now. Could you review again?

File: kibana_objects/client_kibana_objects_export.go [hunk 4]
```diff
@@ -29,16 +29,14 @@ type ExportResults struct {
...
 // https://docs.logz.io/api/#operation/exportSavedObjects
 func (c *KibanaObjectsClient) Export(t exportType) (*ExportResults, error) {
-	exportPayload := struct {
-		T exportType `json:"type"`
-	}{
-		t,
-	}
-
-	payload, err := json.Marshal(&exportPayload)
+	payload, err := json.Marshal(&ExportPayload{t})
```
Comment (yyyogev) [inline kibana_objects/client_kibana_objects_export.go:39]: Much better!

**Review by yyyogev — COMMENTED** [review]
@pengux  Looks good!
The integration test are failing since there is no env variable in your fork for the `LOGZIO_ACCOUNT_ID` and `LOGZIO_API_TOKEN`
Would you rather I'll open a new PR in our repo or you prefer to define those variables in your repo?

**Comment by pengux** [comment]
> @pengux Looks good!
> The integration test are failing since there is no env variable in your fork for the `LOGZIO_ACCOUNT_ID` and `LOGZIO_API_TOKEN`
> Would you rather I'll open a new PR in our repo or you prefer to define those variables in your repo?

I've opened #74 

Closing this PR

**Pull request closed without merging**